src/bundles/files/utils.js

```diff
@@ -1,89 +1,125 @@
+ * 2. `{ status: 'Pending', id: Symbol, state: State }` - State while task is in
+ *    progress. Each yielded value will cause this action and will correspond
+ *    to `state` field.
+ * 3. `{ status: 'Done', id: Symbol, value: Return }` - State when task is
+ *    successfully complete.
+ * 4. `{ status: 'Failed', id: Symbol, error: Error }` - State when task is
+ *    failed do to error.
+ *
+ * @template {string} Name - Name of the task which, correponds to `.type` of
+ * dispatched actions.
+ * @template State - Type of yielded value by a the generator, which will
+ * correspond to `.job.state` of dispatched actions while task is pending.
+ * @template Return - Return type of the task, which will correspond to
+ * `.job.result.value` of dispatched action on succefully completed task.
+ * @template {BundlerContext<State, Job<Name, State, Error, Return>, StoreExt, GetIPFS>} Ctx
+ *
+ * @param {Name} name - Name of the task
+ * @param {(service:IPFSService, store:Ctx) => AsyncGenerator<State, Return, void>} task
+ * @returns {(context:Ctx) => Promise<Return>}
+ */
+export const spawn = (name, task) => async (context) => {
+  // Generate unique id for this task
+  const id = Symbol(name)
+  const type = name
 
-      if (src === store.selectFiles().path) {
-        await store.doUpdateHash(dst)
-      }
+  try {
+    const ipfs = context.getIpfs()
+    if (ipfs == null) {
+      throw Error('IPFS node was not found')
     }
+    context.dispatch({ type, job: { id, status: 'Idle' } })
 
-    // Delete specific logic
-    if (basename === ACTIONS.DELETE) {
-      const src = args[0][0]
-
-      let path = src.split('/')
-      path.pop()
-      path = path.join('/')
-
-      await store.doUpdateHash(path)
+    const process = task(ipfs, context)
+    while (true) {
+      const next = await process.next()
+      if (next.done) {
+        const { value } = next
+        context.dispatch({ type, job: { id, status: 'Done', value } })
+        return value
+      } else {
+        const { value: state } = next
+        context.dispatch({ type, job: { id, status: 'Active', state } })
+      }
     }
   } catch (error) {
-    if (!error.code) {
-      error.code = `ERR_${basename}`
-    }
-
-    console.error(error)
-    dispatch({ type: `FILES_${basename}_FAILED`, payload: { id, error } })
-  } finally {
-    if (basename !== ACTIONS.FETCH) {
-      await store.doFilesFetch()
-    }
-
-    if (basename === ACTIONS.PIN_ADD || basename === ACTIONS.PIN_REMOVE) {
-      await store.doPinsFetch()
-    }
+    context.dispatch({ type, job: { id, status: 'Failed', error } })
+    // Propagate error to a caller.
+    throw error
   }
+}
 
-  return data
+/**
+ * Creates an acton creator that just dispatches given action.
+ * @template T
+ * @param {T} action
+ * @returns {(context:BundlerContext<any, T, any, any>) => Promise<void>}
+ */
+export const send = (action) => async ({ store }) => {
```

Comment:
> What's the point of creating a function that just serves as a way to invoke two variables?

That's fare, but I would go step further and ask what's the point of having redux-bundler with all the actions creators function that returns the function that is passed the context so that it can just dispatch the action ?

```js
const actions = {
  ...,
  doFilesDismissErrors: () => async (context) => context.store.dispatch({
     type: ACTIONS.DISMISS_ERRORS
   }),
   ...
}
```
I just find not having to create intermediate function manually a bit more clear and it reduces boilerplate. 

```js
{
  ...,
  doFilesDismissErrors: () => send({ type: ACTIONS.DISMISS_ERRORS }),
  ...
}
```

That said I do find it style that redux-bundle forces quite a bit annoying.

> Also, I don't see where this function is being used 🤔

It is used quite a bit in the filse/actions.js